tests/test_doc_examples.py

```diff
@@ -11,40 +11,76 @@
 
 
 def get_doc_examples():
     policies = []
-    for key, v in resources().items():
+    for resource_name, v in resources().items():
         for k, cls in itertools.chain(v.filter_registry.items(), v.action_registry.items()):
-            if cls.__doc__:
-                split_doc = [x.split('\n\n') for x in
-                             cls.__doc__.split('yaml')]  # split on yaml and new lines
-                for item in itertools.chain.from_iterable(split_doc):
-                    if 'policies:\n' in item:
-                        policies.append((item, key, cls.__name__))
-
+            if not cls.__doc__:
+                continue
+            # split on yaml and new lines
+            split_doc = [x.split('\n\n') for x in cls.__doc__.split('yaml')]
+            for item in itertools.chain.from_iterable(split_doc):
+                if 'policies:\n' in item:
+                    policies.append((item, resource_name, cls.type))
     return policies
 
 
 class DocExampleTest(BaseTest):
 
+    skip_condition = not (
+        # Okay slightly gross, basically if we're explicitly told via
+        # env var to run doc tests do it.
+        (os.environ.get("C7N_TEST_DOC") in ('yes', 'true') or
+         # Or for ci to avoid some tox pain, we'll auto configure here
+         # to run on the py3.6 test runner, as its the only one
+         # without additional responsibilities.
+         (os.environ.get('C7N_TEST_RUN') and
+          sys.version_info.major == 3 and
+          sys.version_info.minor == 6)))
+
+    @skipif(skip_condition, reason="Doc tests must be explicitly enabled with C7N_DOC_TEST")
     def test_doc_examples(self):
-        errors = []
-        for policy, module, cls_name in get_doc_examples():
-            try:
-                parsed_policy = yaml.safe_load(policy)
-                list(map(lambda p: self.load_policy(p, validate=C7N_TEST_DOCS),
-                         parsed_policy["policies"]))
-            except Exception as e:
-                errors.append((module, cls_name, e))
-
-        assert len(errors) == 0, errors
+        policies = []
+        policy_map = {}
+        idx = 1
+        for ptext, resource_name, el_name in get_doc_examples():
+            data = yaml.safe_load(ptext)
+            for p in data.get('policies', []):
+                # We unique based on name and content to avoid duplicates
+                # from inherited docs.
+                if p['name'] in policy_map:
+                    if p != policy_map[p['name']]:
+                        # Give each policy a unique name with enough
+                        # context that we can identify the origin on
+                        # failures.
+                        p['name'] = "%s-%s-%s-%d" % (
+                            resource_name.split('.')[-1],
+                            el_name,
+                            p.get('name', 'unknown'), idx)
+                    continue
+                policy_map[p['name']] = p
```

Comment:
why not just set the unique name by default?

Reply:
because that won't weed out the duplicates from class inheritance, with inheritance there are 1500+ policies, uniquing on name and content gets us to 365 per my comments on the pr.

Reply:
ah gotcha, thanks.